java/src/jmri/jmrit/logix/Engineer.java

```diff
@@ -1069,19 +1101,22 @@ private void rampDone(boolean stop, String speedType) {
      class ThrottleRamp extends Thread {
 
          private String _endSpeedType;
-         private int _endBlockIdx;   // index of block where down ramp ends.
-                     // not used for up ramps or control commands.
-         private boolean stop = false;    // aborts ramping
-         private boolean ready = false;   // ready for call doRamp
-         private boolean _die = false;    // kills ramp for good
+         private int _endBlockIdx = -1;     // index of block where down ramp ends.
+         private boolean stop = false;      // aborts ramping
+         private boolean _ready = false;    // true when at the wait in doRamp
+         private boolean _rampDown = true;
+         private boolean _die = false;      // kills ramp for good
          private boolean holdRamp = false;
 
          ThrottleRamp() {
             setName("Ramp(" + _warrant.getTrainName() +")");
             _endBlockIdx = -1;
          }
-         boolean isReady() {
-             return ready;
+         final boolean isReady() {
+             return _ready;
+         }
+         final boolean isRampDown() {
+             return _rampDown;
          }
 
          @SuppressFBWarnings(value = "NN_NAKED_NOTIFY", justification="quit is called another thread to clear all ramp waits")
```

Comment:
Is this @SuppressFBWarnings still needed? You removed the `notifyAll`

Reply:
I believe It may be removed.  The notifyAll was moved to the main thread where it needed to be to get termination done right.

Reply:
Could you try to remove the `@SuppressFBWarnings`? Once that's done, I can approve this PR.